package/MDAnalysis/coordinates/FHIAIMS.py

```diff
@@ -0,0 +1,261 @@
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+FHIAIMS file format --- :mod:`MDAnalysis.coordinates.FHIAIMS`
+=============================================================
+
+Classes to read and write _FHIAIMS coordinate files.
+
+Classes
+-------
+
+.. autoclass:: Timestep
+   :members:
+
+.. autoclass:: FHIAIMSReader
+   :members:
+
+.. autoclass:: FHIAIMSWriter
+   :members:
+
+
+Developer notes: ``FHIAIMSWriter`` format strings
+-------------------------------------------------
+
+The :class:`FHIAIMSWriter` class has a :attr:`FHIAIMSWriter.fmt` attribute, which is a dictionary of different
+strings for writing lines in ``.gro`` files.  These are as follows:
+
+``xyz``
+  An atom line without velocities.  Requires that the 'name' and 'pos' keys be supplied.
+  E.g.::
+
+     fmt['xyz'].format(pos=(0.0, 1.0, 2.0), name='O')
+
+``box_orthorhombic``
+  The final line of the gro file which gives box dimensions.  Requires
+  the box keyword to be given, which should be the three cartesian dimensions.
+  E.g.::
+
+     fmt['box_orthorhombic'].format(box=(10.0, 10.0, 10.0))
+
+``box_triclinic``
+  The (optional) initial lines of the file which gives box dimensions. 
+  Requires the box keyword, as a length 9 vector.  This is a flattened
+  version of the (3,3) triclinic vector representation of the unit
+  cell.  
+
+.. _FHIAIMS: https://aimsclub.fhi-berlin.mpg.de/
+
+"""
+from __future__ import absolute_import
+
+import re
+from six.moves import range, zip
+from six import raise_from
+
+import itertools
+import warnings
+
+import numpy as np
+
+from . import base
+from .core import triclinic_box, triclinic_vectors
+from ..exceptions import NoDataError
+from ..lib import util
+from ..lib import mdamath
+
+
+class Timestep(base.Timestep):
+
+    def _init_unitcell(self):
+        return np.zeros((3, 3), dtype=np.float32)
+
+    @property
+    def dimensions(self):
+        """unitcell dimensions (A, B, C, alpha, beta, gamma)
+        """
+        return triclinic_box(self._unitcell[0], self._unitcell[1], self._unitcell[2])
+
+    @dimensions.setter
+    def dimensions(self, new):
+        self._unitcell[:] = triclinic_vectors(new)
+
+
+class FHIAIMSReader(base.SingleFrameReaderBase):
+    """Reader for the FHIAIMS geometry format.
+
+    .. note::
+       Geometry files are single frame-only
+
+    """
+    format = ['IN', 'FHIAIMS']
+    units = {'time': None, 'length': 'Angstrom', 'velocity': None}
+    _Timestep = Timestep
+
+    def _read_first_frame(self):
+        with util.openany(self.filename, 'rt') as fhiaimsfile:
+            lines = fhiaimsfile.readlines()
```

Comment:
I would try not to use `readlines()`.  If you accidentally pass in some multi TB file it will try and parse the entire file then fail without any error message.  Better would be to work through the file line by line, taking what you need and giving a coherent error message when something awry happens.

Reply:
fixed